cpp/include/cudf/io/functions.hpp

```diff
@@ -480,13 +480,17 @@ struct write_parquet_args {
   /// Specify the sink to use for writer output
   sink_info sink;
   /// Specify the compression format to use
-  compression_type compression;
+  compression_type compression = compression_type::AUTO;
   /// Specify the level of statistics in the output file
-  statistics_freq stats_level;
+  statistics_freq stats_level = statistics_freq::STATISTICS_ROWGROUP;
   /// Set of columns to output
   table_view table;
   /// Optional associated metadata
   const table_metadata *metadata;
+  /// Optional raw parquet file metadata output
+  std::vector<uint8_t> *raw_metadata_out = nullptr;
```

Comment:
I think we would need a translation layer that pays the penalty, otherwise we would have to pay the penalty to convert the user-accessible thing to the needed byte vector.

Reply:
The metadata is actually a standard parquet FileMetadata thrift message, so it's presumably possible for python to inspect the contents using existing parquet libraries.